CriteoPublisherSdk/Tests/IntegrationTests/CR_FeedbackMessageFunctionalTests.m

```diff
@@ -281,7 +327,10 @@ - (void)getBidAndWaitForFetchAndFeedbackWithAdUnit:(CRAdUnit *)adUnit {
 
   CR_CacheAdUnit *cacheAdUnit = [CR_AdUnitHelper cacheAdUnitForAdUnit:adUnit];
   [self.criteo getBid:cacheAdUnit
-      responseHandler:^(CR_CdbBid *bid){
+      responseHandler:^(CR_CdbBid *bid) {
+        if (self.config.isLiveBiddingEnabled) {
+          [self.dependencyProvider.feedbackDelegate sendFeedbackBatch];
```

Comment:
Shouldn't this be already done by the bid manager ?

Reply:
Nope, as I've kept the same logic as before, to send metrics with live request, not after, meaning we still have a delay with metrics.
I'd like to change, but a quick attempt showed that wasn't trivial mostly because of asynchronicity, but also because the fetching logic is shared across prefetch and live bid.
What's your take about it? How is this handled on Android side?

Reply:
Ok, understood. On Android CSM are sent in parallel of bid requests. So it depends of this concurrent race :D
In the test we tried to have a deterministic approach by doing N+1 bids to be sure to have N metrics. But it makes test harder to read and to maintain finally.
So up to you. Maybe could you just keep this and add a small comment explaining the why ?

Reply:
So to clarify same thing on iOS with bid and csm are sent in //.

Ok for a comment, I might try to sequence bid then csm FAR as it is the most logical thing to do in live bid mode.
If not, maybe fill an issue for the future?